packages/admin-panel-server/src/viz-builder/dashboardVisualisation/DashboardVisualisationExtractor.ts

```diff
@@ -92,11 +90,22 @@ export class DashboardVisualisationExtractor<
   private vizToReport(previewMode?: PreviewMode): Record<keyof Report, unknown> {
     const { code, permissionGroup, data, presentation } = this.visualisation;
 
-    const fetch = extractFetch(data);
+    const { fetch: vizFetch, aggregate, transform } = data;
+    const validatedVizFetch = yup
+      .object()
+      .required('fetch is a required field')
+      .validateSync(vizFetch);
```

Comment:
Could we instead update `baseVisualisationValidator` to make its validation more thorough?

Reply:
Unfortunately this isn't quite true anymore now that this same class handles legacy vizes which don't have a fetch step...

We could define `baseVizReportValidator` and `baseLegacyVizReportValidator` though? And validate those at the start of `vizToReport()` and `vizToLegacyReport()` respectively?

Reply:
Oh I see, good point! We have validation in a lot of places so I wouldn't worry too much, happy for either what you have here or your above approach